src/components/boundsfitter/BoundsFitter.tsx

```
@@ -5,12 +5,23 @@ type Props = {
+ * Expects `bounds` to be an array of two coordinate pairs, each containing two finite numbers. If the bounds are invalid or malformed, the map view is not changed.
+ */
 export function BoundsFitter(props: Props) {
   const { bounds } = props;
 
   const map = useMap();
   useEffect(() => {
-    map.fitBounds(bounds);
+    // Check if bounds are valid before applying
+    if (bounds && bounds.length === 2 && bounds.every(bound => 
+      bound.length === 2 && 
+      bound.every(coord => typeof coord === 'number' && !isNaN(coord) && isFinite(coord))
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)